aten/src/ATen/native/transformers/cuda/attention_backward.cu

```diff
@@ -380,9 +380,9 @@ _efficient_attention_backward(
     grad_k = chunk.select(2, 1);
     grad_v = chunk.select(2, 2);
   } else {
-    grad_q = at::empty(query.sizes(), query.options());
-    grad_k = at::empty(key.sizes(), key.options());
-    grad_v = at::empty(value.sizes(), value.options());
+    grad_q = at::empty_like(query);
```

Comment:
Pretty sure flash attention used to have the same bug, I guess it was copied and pasted from here and never fixed here.